tuf/ngclient/config.py

```diff
@@ -23,7 +24,13 @@ class UpdaterConfig:
             are used, target download URLs are formed by prefixing the filename
             with a hash digest of file content by default. This can be
             overridden by setting ``prefix_targets_with_hash`` to ``False``.
-
+        offline: When offline, Updater works normally except for two differences: First,
+            expired metadata is considered valid. Second, Updater will not
+            download metadata or targets (if a download would be needed, a DownloadError
+            is raised instead). This is useful for a client that has all required
+            metadata and targets cached locally and is willing to sacrifice timeliness of
```

Comment:
It's hard for a client to know, if they really have all the required metadata and targets cached locally, if they don't consult with the repository and ignore expirations.

Also, while I can see how it is useful for a client to use the metadata and targets cached locally, I am still sceptical about the usefulness of using the TUF client for this. In other words, what is the benefit of using the updater in offline mode vs. not using the updater at all?

We should make this clear somewhere. Probably not here in the docstring.

Reply:
sacrifice timeliness AND compromise recovery